composer.json

```diff
@@ -17,6 +17,7 @@
         "php": "^7.1.3",
         "ext-json": "*",
         "algolia/algoliasearch-client-php": "^2.0",
+        "composer/composer": "^1.8",
```

Comment:
Sorry, buddy. You should wait me to answer before coding. You don't accept new dependencies because this.

Laravel internally have a function get composer's contents. Use that one.

Reply:
No worries - i was keen!

Out of interest, for a package maintainer, why is adding the dependency a bad idea when Laravel also has the same dependency?

`Illuminate\Support\Composer` is not an option as it is a facade to calling composer scripts. The composer path is not exposed as public method.   

Instead we are now using `base_path('composer.json')` and calling `json_decode`.  This is actually better than using `Composer\Composer` because we are definitely loading the laravel project's composer.json, not the scout-extended's.

Reply:
> Out of interest, for a package maintainer, why is adding the dependency a bad idea when Laravel also has the same dependency?

I didn't know this. Anyway, seems way clean now.